src/rt/util/utf.d

```diff
@@ -566,7 +565,7 @@ Checks to see if string is well formed or not. $(D S) can be an array
  of $(D char), $(D wchar), or $(D dchar). Throws a $(D UtfException)
  if it is not. Use to check all untrusted input for correctness.
  */
-void validate(S)(in S s)
+void validate(S)(in S s) pure nothrow @safe @nogc
```

Comment:
I'm still unsure about the whole `ubyte`-for-ASCII idea. I think it's an interesting direction iff we improve Phobos and the language to better support it as well (e.g. allowing comparison of `ubyte[]` and string literals, making `s.startsWith("foo")` work with `ubyte[] s`, etc.)